### DISCUSSION TITLE
Unable to Call start_traj_mode Service

### DISCUSSION THREAD

**Discussion by Saketh7256** [Yaskawa-Global/motoros2#253] [Q&A] [answered]
Hello,

I am experiencing a challenge with the MotoROS2 system while trying to activate trajectory mode on my robot. Initially, I received the following error message:

![Screenshot from 2024-05-17 11-54-22](https://github.com/Yaskawa-Global/motoros2_client_interface_dependencies/assets/167260701/956ecde5-d1a6-497c-996e-256636b5ba6e)

To resolve this, I attempted to activate the trajectory mode using the start_traj_mode service with these commands:

1.ros2 service call /start_traj_mode motoros2_interfaces/srv/StartTrajMode                                        &
2.ros2 service call /start_traj_mode motoros2_interfaces/srv/StartTrajMode "{}"

However, both attempts returned an error stating that "The passed service type is invalid." I confirmed the availability of the /start_traj_mode service using ros2 service list, which shows the service as available.

Could you assist me 
Thank you for your support!

[fjt_client_new.py.zip](https://github.com/Yaskawa-Global/motoros2_client_interface_dependencies/files/15355312/fjt_client_new.py.zip)

**Comment by gavanderhoorn** [comment, marked as answer]
It's possible you forgot to `source /path/to/your/colcon_ws/install/setup.bash` before trying to `ros2 service call` the service.

Could I again ask you to please post your questions about MotoROS2 on the [MotoROS2 Discussion forum](https://github.com/Yaskawa-Global/motoros2/discussions)? Thank you.

**Reply by gavanderhoorn** [reply]
@Saketh7256: have you been able to resolve your issue?

**Reply by ted-miller** [reply]
Through a separate email chain, I can report that he was able to move the robot via the python script. I believe the solution involved sourcing the appropriate workspace(s). 

The current issue is regarding MoveIt. I advised him to open a ticket on the MoveIt discussion forum.
